src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -44,7 +49,41 @@ abstract class ErgoNodeViewHolder[State <: ErgoState[State]](settings: ErgoSetti
   override def postStop(): Unit = {
     log.warn("Stopping ErgoNodeViewHolder")
     history().closeStorage()
-    minimalState().closeStorage
+    minimalState().closeStorage()
+  }
+
+  override def receive: Receive =
+    processRemoteModifiers orElse
+      processLocallyGeneratedModifiers orElse
+      processNewTransactions orElse
+      getCurrentInfo orElse
+      getNodeViewChanges orElse
+      successfulModifier orElse
+      unknownMessage
+
+  private def successfulModifier: Receive = {
+    case SemanticallySuccessfulModifier(mod: ErgoFullBlock) =>
```

Comment:
I think that when you receive `SemanticallySuccessfulModifier` your state might be already on a different height, at least you do not have these guaranties. Test that applies batch of headers `settings.nodeSettings.snapshotCreationInterval`, `settings.nodeSettings.snapshotCreationInterval + 1` and `settings.nodeSettings.snapshotCreationInterval - 1` after that should catch this problem.

Reply:
@catena2w, fixed.